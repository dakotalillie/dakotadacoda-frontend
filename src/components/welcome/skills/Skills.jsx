import React, { Fragment } from 'react';
import { Button, ButtonGroup, Col, Container, Row } from 'reactstrap';

import { tabDisplay } from './utils';
import './skills.css';

class ResumePreview extends React.Component {
  state = {
    active: 'languages'
  };

  setActive = newActive => {
    this.setState({ active: newActive });
  };

  render() {
    const tabs = ['Languages', 'Frameworks', 'Other'];
    const section1 = (
      <Fragment>
        <Col sm="2" className="buttons_col">
          <ButtonGroup vertical>
            {tabs.map(name => (
              <Button
                key={name}
                id={`${name.toLowerCase()}_tab`}
                outline
                block
                color="primary"
                active={this.state.active === name.toLowerCase()}
                onClick={() => this.setActive(name.toLowerCase())}
              >
                {name}
              </Button>
            ))}
          </ButtonGroup>
        </Col>
        <Col sm="6" className="icons_col">
          <Row>{tabDisplay(this.state.active)}</Row>
        </Col>
      </Fragment>
    );

    const section2 = (
      <Col sm="4" className="copy_col">
        <h2>Proficiencies &amp; Skills</h2>
      </Col>
    );

    return (
      <div className="resume_preview section">
        <Container>
          {window.innerWidth > 768 ? (
            <Row>
              {section1}
              {section2}
            </Row>
          ) : (
            <Row>
              {section2}
              {section1}
            </Row>
          )}
        </Container>
      </div>
    );
  }
}

export default ResumePreview;
