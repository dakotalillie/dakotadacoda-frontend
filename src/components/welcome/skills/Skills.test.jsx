import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Skills from './Skills';

describe('Skills', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Skills />);
  });

  it('renders successfully', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('clicking on tabs sets state', () => {
    wrapper.find('#frameworks_tab').simulate('click');
    expect(wrapper.state().active).toEqual('frameworks');
    wrapper.find('#other_tab').simulate('click');
    expect(wrapper.state().active).toEqual('other');
  });
});