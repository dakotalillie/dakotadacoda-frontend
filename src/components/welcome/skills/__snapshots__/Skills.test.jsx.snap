// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills renders successfully 1`] = `
<div
  className="resume_preview section"
>
  <Container
    tag="div"
  >
    <Row
      tag="div"
    >
      <Col
        className="buttons_col"
        sm="2"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <ButtonGroup
          role="group"
          tag="div"
          vertical={true}
        >
          <Button
            active={true}
            block={true}
            color="primary"
            id="languages_tab"
            key="Languages"
            onClick={[Function]}
            outline={true}
            tag="button"
          >
            Languages
          </Button>
          <Button
            active={false}
            block={true}
            color="primary"
            id="frameworks_tab"
            key="Frameworks"
            onClick={[Function]}
            outline={true}
            tag="button"
          >
            Frameworks
          </Button>
          <Button
            active={false}
            block={true}
            color="primary"
            id="other_tab"
            key="Other"
            onClick={[Function]}
            outline={true}
            tag="button"
          >
            Other
          </Button>
        </ButtonGroup>
      </Col>
      <Col
        className="icons_col"
        sm="6"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Row
          tag="div"
        >
          <ItemIcon
            key="JavaScript"
            name="JavaScript"
            src="javascript.png"
            subtitle={true}
            xs="3"
          />
          <ItemIcon
            key="Python"
            name="Python"
            src="python.png"
            subtitle={true}
            xs="3"
          />
          <ItemIcon
            key="Java"
            name="Java"
            src="java.png"
            subtitle={true}
            xs="3"
          />
          <ItemIcon
            key="Ruby"
            name="Ruby"
            src="ruby.png"
            subtitle={true}
            xs="3"
          />
        </Row>
      </Col>
      <Col
        className="copy_col"
        sm="4"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <h2>
          Proficiencies & Skills
        </h2>
      </Col>
    </Row>
  </Container>
</div>
`;
